### PR TITLE
Use implicit logging scope for Activity

### LINKED ISSUE #21020: No need to explicitly hoist Activity inthe HostingLogScope
This is tracking undoing the work done in https://github.com/dotnet/aspnetcore/pull/11211/

When we have [this feature](https://github.com/dotnet/runtime/issues/34305) in the runtime (or more appropriately, the LoggerExternalScopeProvider) to create an implicit scope entry with the current activity, we'll no longer need to do it explicitly in ASP.NET Core.

This is blocked on- https://github.com/dotnet/runtime/issues/34305

### LINKED ISSUE #21966: Enable ActivityTracking in CreateDefaultBuilder
https://github.com/dotnet/runtime/issues/34305 is introducing new API on `LoggerFactoryOptions` to enable `Activity` tracking. We should enable it in `CreateDefaultBuilder`.

Needs to happen with- https://github.com/dotnet/aspnetcore/issues/21020

### PR DESCRIPTION
**Pull request by shirhatti**
This reverts commit f7a2d3c26cfd821d0740fdf5f5ac26845fcd33d0.

Addresses #21020

### CODE REVIEW

File: src/Hosting/Hosting/test/HostingApplicationTests.cs
Comment (Tratcher) [inline src/Hosting/Hosting/test/HostingApplicationTests.cs:86]: Conflict

**Review by Tratcher — CHANGES_REQUESTED** [review]
There are conflicts in the test that need resolving.

File: src/Hosting/Hosting/test/HostingApplicationTests.cs
Comment (shirhatti) [reply src/Hosting/Hosting/test/HostingApplicationTests.cs:102]: Oops. Fixing it

File: src/Hosting/Hosting/test/WebHostTests.cs
Comment (shirhatti) [inline src/Hosting/Hosting/test/WebHostTests.cs:881]: Oops. Need to restore this test

**Review by Tratcher — APPROVED** [review]

**Comment by shirhatti** [comment]
Still waiting on https://github.com/dotnet/aspnetcore/pull/22387 before the new API from runtime is available

**Comment by davidfowl** [comment]
I don’t think I should revert anything. Just make a new change.

**Comment by shirhatti** [comment]
> I don’t think I should revert anything. Just make a new change.

I'm going to squash before merging, so it won't look like a reverted commit when merged. I just found it easier to work this way.

**Comment by tarekgh** [comment]
https://github.com/dotnet/runtime/issues/34305

**Review by tarekgh — APPROVED** [review]

**Comment by tarekgh** [comment]
just wondering do we have any tests covering this functionality in asp.net? I just wanted to ensure we are not introducing any regression there.

**Comment by shirhatti** [comment]
> just wondering do we have any tests covering this functionality in asp.net? I just wanted to ensure we are not introducing any regression there.

We did have some tests that I've removed in the PR. It isn't functionally the same behavior. In place of the 1 scope object that had both Activity and  RequestId/RequestPath. Ideally I wouldn't like to test runtime functionality in this repo.

**Comment by shirhatti** [comment]
@Pilchie Does this need tactics approval? Also, who needs to merge since I don't have access to this branch?

**Comment by tarekgh** [comment]
> We did have some tests that I've removed in the PR. It isn't functionally the same behavior. In place of the 1 scope object that had both Activity and RequestId/RequestPath. Ideally I wouldn't like to test runtime functionality in this repo.

At least did you validate the behavior manually for asp.net?

**Comment by davidfowl** [comment]
Why would it need tactics approaval?

**Comment by shirhatti** [comment]
> Why would it need tactics approval?

We already branched for preview6

**Comment by davidfowl** [comment]
Put it in master, don't rush it into a preview.

**Comment by Pilchie** [comment]
If there is a decent reason to put it in Preview6, I can approve, but I'd like to know why not to wait for Preview7?

**Comment by shirhatti** [comment]
My only motivation for getting this in an earlier preview was that it is a behavior change and getting it out earlier lets us solicit more feedback.

I can re-target this PR if we don't want to rush it into preview6

**Comment by shirhatti** [comment]
@tarekgh Manually verified with console logger. Is there any other logger provider you suggest I verify against given the fact that it's now two different scope objects? 

New log message
```
info: PlaintextApp.Startup[0]
      => SpanId:|50b0a7b9-4f895b67e546d4e6., TraceId:50b0a7b9-4f895b67e546d4e6, ParentId: => ConnectionId:0HM08BPOU3LC2 => RequestPath:/ RequestId:0HM08BPOU3LC2:00000002
      Processing request
```
Old log message

```
info: PlaintextApp.Startup[0]
      => ConnectionId:0HM08BGDIJUGA => RequestPath:/ RequestId:0HM08BGDIJUGA:00000002, SpanId:|c5b2b78c-49b1f71be195b052., TraceId:c5b2b78c-49b1f71be195b052, ParentId:
      Processing request
```

**Comment by davidfowl** [comment]
App insights. I wonder if this should instead go into the host builder’s create default builder

**Comment by shirhatti** [comment]
> I wonder if this should instead go into the host builder’s create default builder

In runtime?https://github.com/dotnet/runtime/blob/96a3bfed4acffbe88403d5f599cee358b880ac5d/src/libraries/Microsoft.Extensions.Hosting/src/Host.cs#L96

I just realized I also need to make the change in here (since we don't use `WebHostBuilder` in the default case) 🤦 https://github.com/dotnet/aspnetcore/blob/5f7e1a0004a8215b1c5b5bfb0f7fac5335f9b942/src/DefaultBuilder/src/WebHost.cs#L209

**Comment by cijothomas** [comment]
> @cijothomas Would this change impact AppInsights? Activity information is no longer present on the `HostingLogScope` object. It's present a separate scope object, however the fidelity of information is preserved.

Don't think so. ApplicationInsights does not directly rely on the activity information from logger. (To correlate with parent request, Activity.Current is used)

File: src/DefaultBuilder/src/WebHost.cs
Comment (davidfowl) [inline src/DefaultBuilder/src/WebHost.cs:224]: I don't think this should be here. This should go into the generic host.

Comment (halter73) [reply src/DefaultBuilder/src/WebHost.cs:224]: This get's called by the generic host. I verified that when doing #22528.

Comment (shirhatti) [reply src/DefaultBuilder/src/WebHost.cs:224]: I thought we agreed to only to enable it for webapps?

Comment (davidfowl) [reply src/DefaultBuilder/src/WebHost.cs:224]: I think we should turn it on by default for all generic hosted apps. That means workers get it for free as well.

Comment (shirhatti) [reply src/DefaultBuilder/src/WebHost.cs:224]: So remove the `ConfigureWebDefaults` changes (make changes to generic host in the runtime repo) and keep the `WebHostBuilder` changes and get this merged?

Comment (tarekgh) [reply src/DefaultBuilder/src/WebHost.cs:224]: > I think we should turn it on by default for all generic hosted apps. That means workers get it for free as well.

Can we clarify the scope of this? I mean how common the apps are using default host against using none-defaulted hosts? I am asking because originally we decided to have the behavior as opt-in to be intentional paying the extra cost when only opting-in .

Comment (davidfowl) [reply src/DefaultBuilder/src/WebHost.cs:224]: Worker service and ASP.NET Core core applications.

Comment (shirhatti) [reply src/DefaultBuilder/src/WebHost.cs:224]: Created a new PR in runtime for this https://github.com/dotnet/runtime/pull/37892

Comment (tarekgh) [reply src/DefaultBuilder/src/WebHost.cs]: do we still need to have this PR?

Comment (Tratcher) [reply src/DefaultBuilder/src/WebHost.cs]: In theory this PR prevents a regression for existing WebHost customers.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**
